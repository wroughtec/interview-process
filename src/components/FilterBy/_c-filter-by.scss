.c-filter-by {
  height: 3rem;
  padding: .5rem .75rem;

  border: 1px solid var(--inputBorder);

  transition: border-color .15s ease-in-out;

  &::placeholder {
    opacity: 1;
  }
}
