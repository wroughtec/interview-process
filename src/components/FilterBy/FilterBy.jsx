// @flow
import React from 'react';
import { nameFilter, cityFilter } from 'consts/filters';
import './_c-filter-by.scss';

type Props = {
  handleChange: Function,
  filterBy: string
};

export const FilterBy = ({ handleChange, filterBy }: Props) => (
  <select className="c-filter-by" value={filterBy} onChange={handleChange}>
    <option value={nameFilter}>Name</option>
    <option value={cityFilter}>City</option>
  </select>
);
