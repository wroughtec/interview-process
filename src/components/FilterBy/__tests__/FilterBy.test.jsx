import React from 'react';
import { shallow } from 'enzyme';
import { FilterBy } from 'components/FilterBy/FilterBy';

describe('FilterBy', () => {
  let component, props;
  beforeEach(() => {
    props = {
      handleChange: jest.fn(),
      filterBy: 'test'
    };

    component = shallow(<FilterBy {...props} />);
  });
  it('should render correctly', () => {
    expect(component).toMatchSnapshot();
  });

  it('on change function should have been called', () => {
    component.find('select').simulate('change', { currentTarget: { value: 'city' } });

    expect(props.handleChange).toHaveBeenCalled();
  });
});
