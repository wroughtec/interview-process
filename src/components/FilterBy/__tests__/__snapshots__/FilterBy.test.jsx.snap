// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterBy should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FilterBy
    filterBy="test"
    handleChange={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <option
          value="name"
        >
          Name
        </option>,
        <option
          value="city"
        >
          City
        </option>,
      ],
      "className": "c-filter-by",
      "onChange": [MockFunction],
      "value": "test",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Name",
          "value": "name",
        },
        "ref": null,
        "rendered": "Name",
        "type": "option",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "City",
          "value": "city",
        },
        "ref": null,
        "rendered": "City",
        "type": "option",
      },
    ],
    "type": "select",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <option
            value="name"
          >
            Name
          </option>,
          <option
            value="city"
          >
            City
          </option>,
        ],
        "className": "c-filter-by",
        "onChange": [MockFunction],
        "value": "test",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Name",
            "value": "name",
          },
          "ref": null,
          "rendered": "Name",
          "type": "option",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "City",
            "value": "city",
          },
          "ref": null,
          "rendered": "City",
          "type": "option",
        },
      ],
      "type": "select",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
